### PR TITLE
Rewrite tests in Jest

### PR DESCRIPTION
**Pull request by illicitonion**
Fixes https://github.com/CodeYourFuture/syllabus/issues/260

### CODE REVIEW

File: mandatory/1-syntax-errors.js [hunk 1]
```diff
@@ -19,24 +19,22 @@ function getTotal(a, b) {
 
 There are some Tests in this file that will help you work out if your code is working.
 
-To run these tests type `node 1-syntax-errors.js` into your terminal
+To run the tests for just this one file, type `npm test -- --testPathPattern 1-syntax-errors` into your terminal
```
Comment (textbook) [inline mandatory/1-syntax-errors.js:22]: Isn't `--testPathPattern` effectively the default for positional args? `npm test -- 1-syntax-errors` would work too.

File: mandatory/1-syntax-errors.js [hunk 2]
```diff
@@ -19,24 +19,22 @@ function getTotal(a, b) {
...
+(Reminder: You must have run `npm install` one time before this will work!)
 
 ===================================================
 */
 
-const util = require('util');
+const {test} = require("@jest/globals");
```
Comment (textbook) [inline mandatory/1-syntax-errors.js:28]:
```suggestion
const {expect, test} = require("@jest/globals");
```

File: mandatory/3-function-output.js [hunk 3]
```diff
@@ -19,37 +19,21 @@ function concatenate(firstWord, secondWord, thirdWord) {
...
-  "concatenate function - case 2 works",
-  concatenate("I", "like", "pizza"),
-  "I like pizza"
-);
-test(
-  "concatenate function - case 3 works",
-  concatenate("I", "am", 13),
-  "I am 13"
-);
+test("concatenate doesn't only accept strings", () => {
```
Comment (textbook) [inline mandatory/3-function-output.js:37]: Good use of test name 👍

File: mandatory/1-syntax-errors.js (hunk 1 above)
Comment (illicitonion) [reply mandatory/1-syntax-errors.js:22]: I thought so too, but when I tried it it still ran all of the tests:

```console
% npm test -- 1-currency-conversion

> javascript-core-1-coursework-week1@1.0.0 test
> jest --testRegex='mandatory[/\\].*\.js$' "1-currency-conversion"

 FAIL  mandatory/2-logic-error.js
[...snipped...]
```

File: mandatory/1-syntax-errors.js (hunk 2 above)
Comment (illicitonion) [reply mandatory/1-syntax-errors.js:28]: This isn't _required_, but I can see the value of adding it for clarity... When `jest` runs individual test closures, it shoves a bunch of stuff into global scope. But I think it probably is worth explicitly importing, even though we haven't taught the trainees anything about `require` yet and are treating this set-up as a completely opaque box :)

**Review by ChrisOwen101 — APPROVED** [review]

**Pull request merged**